komfydb/common/tuple.h

```diff
@@ -45,6 +47,25 @@ class Tuple {
 
   bool operator!=(const Tuple& t) const;
 
+  // Don't use this for empty tuple
+  template <typename H>
+  friend H AbslHashValue(H h, const Tuple& tuple) {
+    for (int i = 0; i < tuple.fields.size(); i++) {
+      Field* field = tuple.fields[i].get();
+      switch (field->GetType().GetValue()) {
+        case Type::INT: {
+          h = H::combine(std::move(h), *dynamic_cast<IntField*>(field));
```

Comment:
Please do not use dynamic_cast, as it checks the types in runtime and throws exceptions. Use static_cast instead (and hope for the best!)